zerver/tests/test_auth_backends.py

```diff
@@ -1505,6 +1507,152 @@ def test_github_oauth2_success_single_email(self) -> None:
+            dict(email="verifiedemail@zulip.com",
+                 verified=True),
+        ]
+        result = self.social_auth_test(account_data_dict,
+                                       subdomain='zulip',
+                                       email_data=email_data,
+                                       expect_choose_email_screen=False,
+                                       next='/user_uploads/image')
+        data = load_subdomain_token(result)
+        self.assertEqual(data['email'], account_data_dict["email"])
+        self.assertEqual(data['name'], self.name)
+        self.assertEqual(data['subdomain'], 'zulip')
+        self.assertEqual(data['next'], '/user_uploads/image')
+        self.assertEqual(result.status_code, 302)
+        parsed_url = urllib.parse.urlparse(result.url)
+        uri = "{}://{}{}".format(parsed_url.scheme, parsed_url.netloc,
+                                 parsed_url.path)
+        self.assertTrue(uri.startswith('http://zulip.testserver/accounts/login/subdomain/'))
+
+    def test_github_oauth2_login_multiple_accounts_exist(self) -> None:
+        # In the login flow, if multiple of the user's verified emails
+        # are associated with existing accounts, we expect the choose
+        # email screen to select which account to use.
+        account_data_dict = dict(email='hamlet@zulip.com', name="Hamlet")
+        email_data = [
+            dict(email=account_data_dict["email"],
+                 verified=True),
+            dict(email='hamlet@zulip.com',
+                 verified=True,
+                 primary=True),
+            dict(email="aaron@zulip.com",
+                 verified=True),
+            dict(email="ignored@example.com",
+                 verified=False),
+        ]
+        result = self.social_auth_test(account_data_dict,
+                                       subdomain='zulip', email_data=email_data,
+                                       expect_choose_email_screen=True,
+                                       next='/user_uploads/image')
+        data = load_subdomain_token(result)
+        self.assertEqual(data['email'], 'hamlet@zulip.com')
+        self.assertEqual(data['name'], 'Hamlet')
+        self.assertEqual(data['subdomain'], 'zulip')
+        self.assertEqual(data['next'], '/user_uploads/image')
+        self.assertEqual(result.status_code, 302)
+        parsed_url = urllib.parse.urlparse(result.url)
+        uri = "{}://{}{}".format(parsed_url.scheme, parsed_url.netloc,
+                                 parsed_url.path)
+        self.assertTrue(uri.startswith('http://zulip.testserver/accounts/login/subdomain/'))
+
+    def test_github_oauth2_login_no_account_exists(self) -> None:
+        # In the login flow, if the user has multiple verified emails,
+        # none of which are associated with an existing account, the
+        # choose email screen should be shown (which will lead to a
+        # "continue to registration" choice).
```

Comment:
I don't understand this test -- the comment I wrote here describes what I'd expect to happen in this case where there are several verified emails and not account, but the code below seems to verify that we end up with logging into an account (I'm not really sure which), not offering to register an account.  What's happening here?